yt_dlp/extractor/cnbc.py

```diff
@@ -1,68 +1,114 @@
+import json
 from .common import InfoExtractor
-from ..utils import smuggle_url
+from ..utils import traverse_obj, unified_strdate, url_or_none
 
 
-class CNBCIE(InfoExtractor):
-    _VALID_URL = r'https?://video\.cnbc\.com/gallery/\?video=(?P<id>[0-9]+)'
-    _TEST = {
-        'url': 'http://video.cnbc.com/gallery/?video=3000503714',
-        'info_dict': {
-            'id': '3000503714',
-            'ext': 'mp4',
-            'title': 'Fighting zombies is big business',
-            'description': 'md5:0c100d8e1a7947bd2feec9a5550e519e',
-            'timestamp': 1459332000,
-            'upload_date': '20160330',
-            'uploader': 'NBCU-CNBC',
+class CNBCVideoIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?cnbc\.com/video/(?:[^/]+/)+(?P<id>[^./?#&]+)\.html'
+
+    _TESTS = [
+        {
+            'url': 'https://www.cnbc.com/video/2023/12/07/mcdonalds-just-unveiled-cosmcsits-new-spinoff-brand.html',
+            'info_dict': {
+                'author': 'Sean Conlon',
+                'channel': 'News Videos',
+                'description': 'McDonald\'s unveiled its new spinoff brand known as CosMc\'s this week, announcing that the new brand\'s first location will open in Bolingbrook, Illinois. CosMc\'s drive-thru focused menu features brand new lemonades and teas, blended beverages, and cold coffee, as well as a small lineup of food. The burger chain first revealed it was creating CosMc\'s as a spinoff during its second-quarter earnings call in July.',
+                'duration': 65.0,
+                'ext': 'mp4',
+                'id': 'mcdonalds-just-unveiled-cosmcsits-new-spinoff-brand',
+                'title': 'Here\'s a first look at McDonald\'s new spinoff brand, CosMc\'s',
+                'thumbnail': 'https://image.cnbcfm.com/api/v1/image/107344192-1701894812493-CosMcsskyHero_2336x1040_hero-desktop.jpg?v=1701894855',
+                'thumbnails': [
+                    {
+                        'url': 'https://image.cnbcfm.com/api/v1/image/107344192-1701894812493-CosMcsskyHero_2336x1040_hero-desktop.jpg?v=1701894855'
+                    }
+                ],
```

Comment:
No need to specify thumbnails like this.

Just let the test data be generated by running the tests without the data. All missing fields will be displayed in the error message.